### ISSUE TITLE
Make `cls.__class__` resolve to `type` rather than `property`

### ISSUE THREAD

**Issue by Glinte** [microsoft/pyright#10735]
[playground](https://pyright-play.net/?code=MYGwhgzhAECCBcAoaLoAdIUYgtgUwBcxQJ5oCBPNPaAXjgDoB9J0TFxIA)

```python
class A:
    pass

metacls: type = A.__class__
#               ^^^^^^^^^^^
# Type "property" is not assignable to declared type "type"
#   "property" is not assignable to "type"  (reportAssignmentType)
```

This seems to be because pyright considers a class to first be `type[object]`, and only secondly as `object`, which does make sense but causes this edge case.

**Comment by erictraut** [comment]
I think pyright's behavior here is correct given the type information in the typeshed stubs. It's also consistent with behavior of other type checkers. I generally try not to override the type information found in the typeshed stubs. If you'd like to suggest a change, please file a bug or PR in the typeshed project.

**Comment by Glinte** [comment]
I believe the typeshed types are correct, it is the fact that pyright is not considering the fact that `A` can be both `type[object]` (very handwavy, A is not actually `type[object]`, just `type`) AND `object` at the same time. This unique behavior of `object` is the culprit. I do not think of the potential fix as overriding the typeshed stubs, but rather making pyright acknowledge classes as `object`s and adding `object` methods to classes.

Pyright does already recognize classes as instances of `object` in an `isinstance` check. But I guess the behaviour of `object` and classes is highly special-cased, since even though a class is technically an object, methods from `object` like `cls.__eq__` does not return a bound method. Such that actually fully considering classes as objects would create much more issues that it would solve.
